### ISSUE TITLE
mysql plugin field overflow

### ISSUE THREAD

**Issue by lassizci** [influxdata/telegraf#515]
Hi,

is there a way to collect from mysql and output to influxdb? Currently I seem to get:

```
Jan 14 19:17:50 mysql1 telegraf[6240]: 2016/01/14 18:17:50 Gathered metrics, ({10s} interval), from 10 plugins in 47.416895ms
Jan 14 19:17:50 mysql1 telegraf[6240]: 2016/01/14 18:17:50 ERROR: {"error":"write failed: field overflow"}
Jan 14 19:17:50 mysql1 telegraf[6240]: 2016/01/14 18:17:50 Error in output [influxdb]: Could not write to any InfluxDB server in cluster, retrying in 10s
Jan 14 19:17:50 mysql1 telegraf[6240]: 2016/01/14 18:17:50 ERROR: {"error":"write failed: field overflow"}
Jan 14 19:17:50 mysql1 telegraf[6240]: 2016/01/14 18:17:50 Error in output [influxdb]: Could not write to any InfluxDB server in cluster, retrying in 10s
Jan 14 19:17:50 mysql1 telegraf[6240]: 2016/01/14 18:17:50 ERROR: {"error":"write failed: field overflow"}
Jan 14 19:17:50 mysql1 telegraf[6240]: 2016/01/14 18:17:50 FATAL: Write to output [influxdb] failed 3 times, dropping 132 metrics
```

telegraf with -test outputs the statistics though:
```
Plugin: mysql, Collection 1
> mysql,server=mysql1.****.**:3306 aborted_clients=9244i,aborted_connects=574i,bytes_received=10663307448899i,bytes_sent=17070670007775i,commands_admin_commands=61i,commands_alter_db=0i,commands_alter_db_upgrade=0i,commands_alter_event=0i,commands_alter_function=0i,commands_alter_procedure=0i,commands_alter_server=0i,commands_alter_table=0i,commands_alter_tablespace=0i,commands_analyze=0i,commands_assign_to_keycache=0i,commands_begin=0i,commands_binlog=0i,commands_call_procedure=0i,commands_change_db=11i,commands_change_master=0i,commands_check=0i,commands_checksum=0i,commands_commit=0i,commands_compound_sql=0i,commands_create_db=0i,commands_create_event=0i,commands_create_function=0i,commands_create_index=0i,commands_create_procedure=0i,commands_create_role=0i,commands_create_server=0i,commands_create_table=0i,commands_create_temporary_table=0i,commands_create_trigger=0i,commands_create_udf=0i,commands_create_user=0i,commands_create_view=0i,commands_dealloc_sql=0i,commands_delete=0i,commands_delete_multi=0i,commands_do=0i,commands_drop_db=0i,commands_drop_event=0i,commands_drop_function=0i,commands_drop_index=0i,commands_drop_procedure=0i,commands_drop_role=0i,commands_drop_server=0i,commands_drop_table=0i,commands_drop_temporary_table=0i,commands_drop_trigger=0i,commands_drop_user=0i,commands_drop_view=0i,commands_empty_query=0i,commands_execute_sql=0i,commands_flush=38i,commands_get_diagnostics=0i,commands_grant=0i,commands_grant_role=0i,commands_ha_close=0i,commands_ha_open=0i,commands_ha_read=0i,commands_help=0i,commands_insert=17740612518i,commands_insert_select=0i,commands_install_plugin=0i,commands_kill=35i,commands_load=0i,commands_lock_tables=0i,commands_optimize=0i,commands_preload_keys=0i,commands_prepare_sql=0i,commands_purge=0i,commands_purge_before_date=901i,commands_release_savepoint=0i,commands_rename_table=0i,commands_rename_user=0i,commands_repair=0i,commands_replace=0i,commands_replace_select=0i,commands_reset=0i,commands_resignal=0i,commands_revoke=0i,commands_revoke_all=0i,commands_revoke_role=0i,commands_rollback=0i,commands_rollback_to_savepoint=0i,commands_savepoint=0i,commands_select=28389575i,commands_set_option=7i,commands_show_authors=0i,commands_show_binlog_events=0i,commands_show_binlogs=0i,commands_show_charsets=0i,commands_show_collations=0i,commands_show_contributors=0i,commands_show_create_db=0i,commands_show_create_event=0i,commands_show_create_func=0i,commands_show_create_proc=0i,commands_show_create_table=7i,commands_show_create_trigger=0i,commands_show_databases=14i,commands_show_engine_logs=0i,commands_show_engine_mutex=0i,commands_show_engine_status=0i,commands_show_errors=0i,commands_show_events=0i,commands_show_explain=0i,commands_show_fields=76i,commands_show_function_status=0i,commands_show_generic=0i,commands_show_grants=2i,commands_show_keys=1i,commands_show_master_status=0i,commands_show_open_tables=0i,commands_show_plugins=0i,commands_show_privileges=0i,commands_show_procedure_status=0i,commands_show_processlist=1i,commands_show_profile=0i,commands_show_profiles=0i,commands_show_relaylog_events=0i,commands_show_slave_hosts=0i,commands_show_slave_status=0i,commands_show_status=28i,commands_show_storage_engines=0i,commands_show_table_status=10i,commands_show_tables=15i,commands_show_triggers=0i,commands_show_variables=9i,commands_show_warnings=0i,commands_shutdown=0i,commands_signal=0i,commands_start_all_slaves=0i,commands_start_slave=0i,commands_stmt_close=0i,commands_stmt_execute=0i,commands_stmt_fetch=0i,commands_stmt_prepare=0i,commands_stmt_reprepare=0i,commands_stmt_reset=0i,commands_stmt_send_long_data=0i,commands_stop_all_slaves=0i,commands_stop_slave=0i,commands_truncate=0i,commands_uninstall_plugin=0i,commands_unlock_tables=0i,commands_update=0i,commands_update_multi=0i,commands_xa_commit=0i,commands_xa_end=0i,commands_xa_prepare=0i,commands_xa_recover=0i,commands_xa_rollback=0i,commands_xa_start=0i,created_tmp_disk_tables=270500i,created_tmp_files=702723i,created_tmp_tables=28565230i,handler_commit=35509613449i,handler_delete=0i,handler_discover=0i,handler_external_lock=0i,handler_icp_attempts=956900283i,handler_icp_match=205020259i,handler_mrr_init=0i,handler_mrr_key_refills=0i,handler_mrr_rowid_refills=0i,handler_prepare=35481225034i,handler_read_first=3i,handler_read_key=374911865799i,handler_read_last=2i,handler_read_next=3445846384413i,handler_read_prev=15085574i,handler_read_rnd=101877038139i,handler_read_rnd_deleted=0i,handler_read_rnd_next=91069665304i,handler_rollback=66i,handler_savepoint=0i,handler_savepoint_rollback=0i,handler_tmp_update=390429590484i,handler_tmp_write=150587259363i,handler_update=3549866966i,handler_write=17740612517i,innodb_available_undo_logs=128i,innodb_background_log_sync=3241580i,innodb_buffer_pool_bytes_data=83996721152i,innodb_buffer_pool_bytes_dirty=952500224i,innodb_buffer_pool_dump_status=0i,innodb_buffer_pool_load_status=0i,innodb_buffer_pool_pages_data=5126753i,innodb_buffer_pool_pages_dirty=58136i,innodb_buffer_pool_pages_flushed=610219194i,innodb_buffer_pool_pages_free=8175i,innodb_buffer_pool_pages_lru_flushed=0i,innodb_buffer_pool_pages_made_not_young=2419821941i,innodb_buffer_pool_pages_made_young=463521629i,innodb_buffer_pool_pages_misc=107944i,innodb_buffer_pool_pages_old=1892330i,innodb_buffer_pool_pages_total=5242872i,innodb_buffer_pool_read_ahead=155423i,innodb_buffer_pool_read_ahead_evicted=0i,innodb_buffer_pool_read_ahead_rnd=0i,innodb_buffer_pool_read_requests=677424495489i,innodb_buffer_pool_reads=140483486i,innodb_buffer_pool_wait_free=0i,innodb_buffer_pool_write_requests=28601480505i,innodb_checkpoint_age=209778641i,innodb_checkpoint_max_age=434154333i,innodb_data_fsyncs=3734851123i,innodb_data_pending_fsyncs=1i,innodb_data_pending_reads=0i,innodb_data_pending_writes=0i,innodb_data_read=2571816816640i,innodb_data_reads=156965297i,innodb_data_writes=4288802944i,innodb_data_written=23913731554816i,innodb_dblwr_pages_written=610219194i,innodb_dblwr_writes=21779748i,innodb_deadlocks=0i,innodb_defragment_compression_failures=0i,innodb_defragment_count=0i,innodb_defragment_failures=0i,innodb_encryption_rotation_estimated_iops=0i,innodb_encryption_rotation_pages_flushed=0i,innodb_encryption_rotation_pages_modified=0i,innodb_encryption_rotation_pages_read_from_cache=0i,innodb_encryption_rotation_pages_read_from_disk=0i,innodb_have_atomic_builtins=0i,innodb_have_bzip2=0i,innodb_have_lz4=0i,innodb_have_lzma=0i,innodb_have_lzo=0i,innodb_have_snappy=0i,innodb_history_list_length=958i,innodb_ibuf_discarded_delete_marks=0i,innodb_ibuf_discarded_deletes=0i,innodb_ibuf_discarded_inserts=0i,innodb_ibuf_free_list=10542i,innodb_ibuf_merged_delete_marks=3318i,innodb_ibuf_merged_deletes=547i,innodb_ibuf_merged_inserts=85464730i,innodb_ibuf_merges=16506102i,innodb_ibuf_segment_size=10687i,innodb_ibuf_size=144i,innodb_log_waits=0i,innodb_log_write_requests=4104674693i,innodb_log_writes=3653112296i,innodb_lsn_current=4607329360123i,innodb_lsn_flushed=4607329359607i,innodb_lsn_last_checkpoint=4607119581482i,innodb_master_thread_active_loops=3240781i,innodb_master_thread_idle_loops=894i,innodb_max_trx_id=29899788585i,innodb_mem_adaptive_hash=3128542320i,innodb_mem_dictionary=340095538i,innodb_mem_total=88583700480i,innodb_mutex_os_waits=20451618i,innodb_mutex_spin_rounds=6318926460i,innodb_mutex_spin_waits=3140027227i,innodb_num_index_pages_written=598029249i,innodb_num_non_index_pages_written=3690704746i,innodb_num_page_compressed_trim_op=0i,innodb_num_page_compressed_trim_op_saved=0i,innodb_num_pages_decrypted=0i,innodb_num_pages_encrypted=0i,innodb_num_pages_page_compressed=0i,innodb_num_pages_page_compression_error=575196932i,innodb_num_pages_page_decompressed=0i,innodb_oldest_view_low_limit_trx_id=0i,innodb_onlineddl_pct_progress=0i,innodb_onlineddl_rowlog_pct_used=0i,innodb_onlineddl_rowlog_rows=0i,innodb_os_log_fsyncs=3653674569i,innodb_os_log_pending_fsyncs=1i,innodb_os_log_pending_writes=0i,innodb_os_log_written=3918334406144i,innodb_page_compression_saved=0i,innodb_page_compression_trim_sect1024=0i,innodb_page_compression_trim_sect16384=0i,innodb_page_compression_trim_sect2048=0i,innodb_page_compression_trim_sect32768=0i,innodb_page_compression_trim_sect4096=0i,innodb_page_compression_trim_sect512=0i,innodb_page_compression_trim_sect8192=0i,innodb_page_size=16384i,innodb_pages_created=5294727i,innodb_pages_read=156963069i,innodb_pages_written=610219194i,innodb_purge_trx_id=29899788575i,innodb_purge_undo_no=0i,innodb_read_views_memory=5680i,innodb_row_lock_current_waits=0i,innodb_row_lock_time=388804i,innodb_row_lock_time_avg=1i,innodb_row_lock_time_max=456i,innodb_row_lock_waits=231658i,innodb_rows_deleted=0i,innodb_rows_inserted=181760439i,innodb_rows_read=3479545613730i,innodb_rows_updated=3549866968i,innodb_s_lock_os_waits=22079997i,innodb_s_lock_spin_rounds=735059826i,innodb_s_lock_spin_waits=34202675i,innodb_scrub_background_page_reorganizations=0i,innodb_scrub_background_page_split_failures_missing_index=0i,innodb_scrub_background_page_split_failures_out_of_filespace=0i,innodb_scrub_background_page_split_failures_underflow=0i,innodb_scrub_background_page_split_failures_unknown=0i,innodb_scrub_background_page_splits=0i,innodb_secondary_index_triggered_cluster_reads=220304424i,innodb_secondary_index_triggered_cluster_reads_avoided=0i,innodb_system_rows_deleted=0i,innodb_system_rows_inserted=0i,innodb_system_rows_read=0i,innodb_system_rows_updated=0i,innodb_truncated_status_writes=0i,innodb_x_lock_os_waits=24531726i,innodb_x_lock_spin_rounds=936319578i,innodb_x_lock_spin_waits=19397429i,key_blocks_not_flushed=0i,key_blocks_unused=214334i,key_blocks_used=0i,key_blocks_warm=0i,key_read_requests=0i,key_reads=0i,key_write_requests=0i,key_writes=0i,open_files=61i,open_streams=0i,open_table_definitions=34i,open_tables=255i,opened_files=2398293i,opened_plugin_libraries=0i,opened_table_definitions=16i,opened_tables=4640i,opened_views=0i,qcache_free_blocks=1i,qcache_free_memory=16759656i,qcache_hits=1i,qcache_inserts=44i,qcache_lowmem_prunes=0i,qcache_not_cached=28389532i,qcache_queries_in_cache=0i,qcache_total_blocks=1i,queries=17869631228i,slow_queries=0i,table_locks_immediate=17762792217i,table_locks_waited=0i,threads_cached=56i,threads_connected=471i,threads_created=2787i,threads_running=4i 1452795199247078028
> mysql_users,server=mysql1.****.**:3306,user=stats connections=470i 1452795199250504591
```

so that is probably more than influxdb can handle.

**Comment by sparrc** [comment]
@lassizci This is because InfluxDB 0.9.x is limited to 255 fields per measurement.

The simple fix would be to use the InfluxDB [0.10.0 beta](https://influxdata.com/downloads/), or use Telegraf 0.2.4.

**Comment by lassizci** [comment]
Allright, thanks for the info!
